src/gsy_e/models/strategy/ev_charger.py

```diff
@@ -0,0 +1,139 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of Grid Singularity Exchange.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from logging import getLogger
+from typing import Optional, Union, Dict
+
+from pendulum import DateTime
+from gsy_framework.validators import EVChargerValidator
+from gsy_framework.enums import GridIntegrationType, EVChargerStatus
+from gsy_framework.constants_limits import ConstSettings, GlobalConfig
+from gsy_framework.read_user_profile import UserProfileReader, InputProfileTypes
+
+from gsy_e.models.strategy.state.evcharger_state import EVChargerState, EVChargingSession
+from gsy_e.models.strategy.storage import StorageStrategy
+
+log = getLogger(__name__)
+
+
+EVChargerSettings = ConstSettings.EVChargerSettings
+StorageSettings = ConstSettings.StorageSettings
+
+
+class EVChargerStrategy(StorageStrategy):
+    """Strategy class EV Charger. Similar to StorageStrategy but only during active sessions."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        grid_integration: GridIntegrationType = GridIntegrationType.BIDIRECTIONAL,
+        charging_sessions: list[EVChargingSession] = [],
+        maximum_power_rating_kW: float = ConstSettings.EVChargerSettings.MAX_POWER_RATING_KW,
+        preferred_charging_power: Optional[Union[float, Dict[DateTime, float]]] = None,
+        **kwargs,
+    ):
+        """
+        Args:
+             grid_integration: connection between the grid and EVs
+             preferred_charging_power: User-specified charging power (kW) per market slot that
+                replaces default strategy. Can be a constant value or time-based profile.
+        """
+        EVChargerValidator.validate(
+            grid_integration=grid_integration,
+            maximum_power_rating_kW=maximum_power_rating_kW,
+            preferred_charging_power=preferred_charging_power,
+        )
+
+        super().__init__(**kwargs)
+
+        self.grid_integration = grid_integration
+        self.maximum_power_rating_kW = maximum_power_rating_kW
+        self.preferred_charging_power = preferred_charging_power
+        self.charging_sessions = sorted(charging_sessions, key=lambda s: s.plug_in_time)
```

Comment:
Maybe not for this initial version, but I think we need a validator that checks if the charging sessions do not overlap and that the plug_in_time follows the slot times. What do you think? Should we add it even now?